x-pack/plugins/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout_internal.test.tsx

```diff
@@ -0,0 +1,745 @@
+
+      const copyButton = findTestSubject(wrapper, 'sts-copy-link'); // this link is only present in the warning callout
+
+      await act(async () => {
+        copyButton.simulate('click');
+        await nextTick();
+        wrapper.update();
+      });
+
+      expect(wrapper.find(CopySavedObjectsToSpaceFlyout)).toHaveLength(1);
+      expect(onClose).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('without enableCreateNewSpaceLink', () => {
+    it('does not render a NoSpacesAvailable component when no spaces are available', async () => {
+      const { wrapper, onClose } = await setup({
+        mockSpaces: [{ id: 'my-active-space', name: 'my active space', disabledFeatures: [] }],
+      });
+
+      expect(wrapper.find(ShareToSpaceForm)).toHaveLength(1);
+      expect(wrapper.find(EuiLoadingSpinner)).toHaveLength(0);
+      expect(wrapper.find(NoSpacesAvailable)).toHaveLength(0);
+      expect(onClose).toHaveBeenCalledTimes(0);
+    });
+
+    it('does not render a NoSpacesAvailable component when only the active space is available', async () => {
+      const { wrapper, onClose } = await setup({
+        mockSpaces: [{ id: 'my-active-space', name: '', disabledFeatures: [] }],
+      });
+
+      expect(wrapper.find(ShareToSpaceForm)).toHaveLength(1);
+      expect(wrapper.find(EuiLoadingSpinner)).toHaveLength(0);
+      expect(wrapper.find(NoSpacesAvailable)).toHaveLength(0);
+      expect(onClose).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('with enableCreateNewSpaceLink', () => {
+    const enableCreateNewSpaceLink = true;
+
+    it('renders a NoSpacesAvailable component when no spaces are available', async () => {
+      const { wrapper, onClose } = await setup({
+        enableCreateNewSpaceLink,
+        mockSpaces: [{ id: 'my-active-space', name: 'my active space', disabledFeatures: [] }],
+      });
+
+      expect(wrapper.find(ShareToSpaceForm)).toHaveLength(1);
+      expect(wrapper.find(EuiLoadingSpinner)).toHaveLength(0);
+      expect(wrapper.find(NoSpacesAvailable)).toHaveLength(1);
+      expect(onClose).toHaveBeenCalledTimes(0);
+    });
+
+    it('renders a NoSpacesAvailable component when only the active space is available', async () => {
+      const { wrapper, onClose } = await setup({
+        enableCreateNewSpaceLink,
+        mockSpaces: [{ id: 'my-active-space', name: '', disabledFeatures: [] }],
+      });
+
+      expect(wrapper.find(ShareToSpaceForm)).toHaveLength(1);
+      expect(wrapper.find(EuiLoadingSpinner)).toHaveLength(0);
+      expect(wrapper.find(NoSpacesAvailable)).toHaveLength(1);
+      expect(onClose).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  it('handles errors thrown from shareSavedObjectsAdd API call', async () => {
+    const { wrapper, mockSpacesManager, mockToastNotifications } = await setup();
+
+    mockSpacesManager.shareSavedObjectAdd.mockImplementation(() => {
+      return Promise.reject(Boom.serverUnavailable('Something bad happened'));
+    });
```

Comment:
__nit:__ (here and in another case below)
```suggestion
    mockSpacesManager.shareSavedObjectAdd.mockRejectedValue(
      Boom.serverUnavailable('Something bad happened')
    );
```